### PR TITLE
[IST-32] Address Set-Env Defect in @actions/core

### PR DESCRIPTION
**Pull request by twistedpair**
- Update to latest `@actions/core`
- Use public NPM repo `package-lock.json`
- Include Typescript compiler as dev dependency to allow building release
- Bump version in `package.json`
- Flush out type system
- Handle `workspace` or `workspace` (docs pointed to one, but it wasn't implemented), and `workspace-id` (GitHub Actions convention)

### CODE REVIEW

**Comment by jbaldassari** [comment]
`node_modules` has to be checked in?

**Review by jbaldassari — APPROVED** [review]
Hard to tell what to review here, but it looks like just dependency changes.  Looks fine assuming the `node_modules` dependencies are supposed to be checked in.

**Comment by twistedpair** [comment]
Yes, that is correct.

**Comment by twistedpair** [comment]
This is the result of running the release script `npm run release`

**Comment by gcooney** [comment]
@twistedpair I've been trying to run npm run release on the release branches only (see CONTRIBUTING.md) so we don't pollute the PRs with compiled files.  I think the other github action actually removed these files all together.

**Comment by twistedpair** [comment]
Thanks, @gcooney . My error. I'll update the README. Looks like the last two releases were not done from the release branch. I'll update this PR to remove compiled files and keep it simple.

File: src/index.ts [hunk 1]
```diff
@@ -2,18 +2,23 @@ import * as core from '@actions/core';
...
 import * as toolCache from '@actions/tool-cache';
 
+type Option<T> = T | undefined;
+
 async function run() {
-  const version = core.getInput('version', {required: false});
-  const workspace = core.getInput('workspace', {required: false});
-  const apiKey: string | undefined = process.env.MABL_API_KEY;
+  const version: Option<string> = core.getInput('version', {required: false});
+  // Allow new or old syntax - some docs said 'workspace', others said 'workspace_id'
```
Comment (twistedpair) [inline src/index.ts:9]: Updated argument handling, and flushed out types where missing.

**Pull request merged**